test/server/admin/clusters_params_test.cc

```diff
@@ -0,0 +1,64 @@
+#include "source/common/buffer/buffer_impl.h"
+#include "source/server/admin/clusters_params.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace Server {
+namespace {
+
+TEST(ClustersParamsTest, FormatDefaultsToTextWhenNotSupplied) {
+  ClustersParams params;
+  Buffer::OwnedImpl buffer;
+  std::string url = "localhost:1337/clusters";
+  ClustersParams::Format expected_format = ClustersParams::Format::Text;
+  Http::Code expected_code = Http::Code::OK;
+
+  Http::Code code = params.parse(url, buffer);
+
+  EXPECT_EQ(params.format_, expected_format);
+  EXPECT_EQ(code, expected_code);
+}
+
+TEST(ClustersParamsTest, FormatSetToTextWhenTextSupplied) {
+  ClustersParams params;
+  Buffer::OwnedImpl buffer;
+  std::string url = "localhost:1337/clusters?format=text";
+  ClustersParams::Format expected_format = ClustersParams::Format::Text;
+  Http::Code expected_code = Http::Code::OK;
+
+  Http::Code code = params.parse(url, buffer);
+
+  EXPECT_EQ(params.format_, expected_format);
+  EXPECT_EQ(code, expected_code);
+}
+
+TEST(ClustersParamsTest, FormatSetToJsonWhenJsonSupplied) {
+  ClustersParams params;
+  Buffer::OwnedImpl buffer;
+  std::string url = "localhost:1337/clusters?format=json";
+  ClustersParams::Format expected_format = ClustersParams::Format::Json;
+  Http::Code expected_code = Http::Code::OK;
+
+  Http::Code code = params.parse(url, buffer);
+
+  EXPECT_EQ(params.format_, expected_format);
+  EXPECT_EQ(code, expected_code);
```

Comment:
nit: WDYT of reducing line-count per test with a pattern like:
```
ClustersParams params;
Buffer::OwnedImpl buffer;
EXPECT_EQ(Http::Code::OK, params.parse("localhost:1337/clusters?format=json", buffer));
EXPECT_EQ(ClustersParams::Format::Json, params.format_);
EXPECT_EQ("", buffer.toString()) << "no error message";
```
What you have is fine; was just trying to reduce the amount of text to read per case.